internal/photoprism/mediafile.go

```diff
@@ -332,6 +332,17 @@ func (m *MediaFile) ExtractEmbeddedVideo() (string, error) {
 
 	// Get the embedded video field name from the file metadata.
 	if metaData := m.MetaData(); metaData.Error == nil && metaData.EmbeddedVideo != "" {
+		// Check if an embedded video sidecar already exists
+		if embeddedVideo := m.HasEmbeddedVideo(); embeddedVideo != "" {
```

Comment:
Although the related MP4 video may also be in the originals path, e.g. if it was extracted manually or if the originals path is the same as the sidecar path, `ExtractEmbeddedVideo()` should only return a filename here if the MP4 is in the sidecar path, since `MediaFile.RelatedFiles()` will automatically find it in the originals path. Otherwise, if we return the name of any MP4 sidecar file, the same file may appear twice in `result.Files` - which would be wrong. Alternatively the handling in `RelatedFiles()` must be modified.

If this becomes too complex and confusing, just ignore it and I'll help you when I get back!